test/migrators/module/namespace_references/loop_error.hrx

```diff
@@ -0,0 +1,21 @@
+<==> arguments
```

Comment:
While it makes sense to error in a case like this, it's not really the core of the issue in #137, since the code does not compile as-is, since `$var` is referenced before it's defined.

We specifically care about cases where the code compiles before migration due to an implicit dependency, but creates a loop when we try to migrate it to `@use` (see the example in Natalie's original comment on #137).

Reply:
I reworded the error to give fixing options rather than just pointing out the error, also added Natalie's example in the tests.